test/fixtures/privileges.sql

```diff
@@ -62,9 +62,12 @@ GRANT USAGE ON SEQUENCE
 TO postgrest_test_anonymous;
 
 -- Privileges for non anonymous users
+GRANT USAGE ON SCHEMA test TO postgrest_test_default_role;
 GRANT USAGE ON SCHEMA test TO postgrest_test_author;
 GRANT ALL ON TABLE authors_only TO postgrest_test_author;
 
 GRANT SELECT (article_id, user_id) ON TABLE limited_article_stars TO postgrest_test_anonymous;
 GRANT INSERT (article_id, user_id) ON TABLE limited_article_stars TO postgrest_test_anonymous;
 GRANT UPDATE (article_id, user_id) ON TABLE limited_article_stars TO postgrest_test_anonymous;
+
+ALTER ROLE postgrest_test_authenticator set search_path = 'public', 'test';
```

Comment:
Maybe if you write the tests i will understand better but until now everything is kind of built not not depend on the search schema at all, every component in the query is fully qualified, and search schema should not play a role anywhere. It's especially weird that the search path for the authenticator plays a role since this is a user that should work with absolutley no privileges, or settings.

Maybe add the tests as you have it now then i can better understand what is going on and how to remove this search path dependency